.github/workflows/verify.yaml

```diff
@@ -427,6 +422,13 @@ jobs:
   Check-Cloud-Environments:
     environment: Azure-dev
     runs-on: ubuntu-latest
+
+    # Grant permissions to obtain federated identity credentials
+    # see https://docs.github.com/actions/deployment/security-hardening-your-deployments/configuring-openid-connect-in-azure
+    permissions:
+      id-token: write
+      contents: read
+
```

Comment:
These permissions are required for reading secrets during `Azure-Cloud-Integration-Test` job.

So maybe we can leave these ones at the top same as before and just add the additional required ones to `Upload-Test-Report`.

Reply:
@cpeeyush since these are required only by `Azure-Cloud-Integration-Test` job why don't add them only on that job? 
Setting these globally caused the issue, using global settings is not a good practice since these permissions are needed by 2 jobs out of 20.

Reply:
Sure, agree. Let's try this on main once it merged to see if we see any failure. As @paullatzelsperger mentioned. Thanks.